### PR TITLE
Add fetch user segments

### PR DESCRIPTION
**Pull request by MikeMcElroy**
Adding in API to access user segmentation API for CDP use cases

### CODE REVIEW

File: lib/plugins/personalization.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  errorCallback = errorCallback || noop
+
+  invariant(
+    typeof audienceToken === 'string',
+    'audienceToken must be a string; received "' + audienceToken.toString() + '"'
+  )
+
+  jsonp('https://cdp.in.treasuredata.com/cdp/lookup/collect/segments?token=' + audienceToken, {
+    prefix: 'TreasureJSONPCallback',
+    timeout: 10000
```
Comment (dzannotti) [inline lib/plugins/personalization.js]: I think we should unify this (`globalid` uses this too), and possibly make it configurable (some clients might need different timeout settings)

File: lib/plugins/personalization.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  }, function (err, res) {
+    return err ? errorCallback(err) : successCallback(res && res.key, res && res.values)
```
Comment (dzannotti) [inline lib/plugins/personalization.js]: Maybe i'm being massively picky here, feel free to disregard this. Other `successCallback` return 1 parameter (the whole response), might be worth investigating and bringing all on the same rule

File: lib/plugins/personalization.js (hunk 1 above)
Comment (MikeMcElroy) [inline lib/plugins/personalization.js]: Unify/configurize these settings?  I'm kind of against that until we see a real need to do so.

File: lib/plugins/personalization.js (hunk 2 above)
Comment (MikeMcElroy) [inline lib/plugins/personalization.js]: We only return the response for `addRecord`'s `successCallback`.  The `fetchGlobalId` API returns the `request['global_id']` to the callback, so I think we should keep to that.

**Comment by dzannotti** [comment]
:+1:

**Pull request merged**